src/Numerical_aux/create_fault_rotationmatrix.f90

```diff
@@ -90,10 +90,14 @@ SUBROUTINE create_fault_rotationmatrix(rotmat,iFace,EQN,MESH,iRotmat)
     n = MESH%Fault%geoNormals(1:3,iFace)
     ! s = MESH%Fault%geoTangent1(1:3,iFace)
     ! t = MESH%Fault%geoTangent2(1:3,iFace)
-
     strike_vector(1) = n(2)/sqrt(n(1)**2+n(2)**2)
     strike_vector(2) = -n(1)/sqrt(n(1)**2+n(2)**2)
     strike_vector(3) = 0.0D0
+    
+    !convention for flat faults
+    if (norm2(strike_vector).eq.0) then
```

Comment:
Does this actually work in practice? I'd expect something like `(norm2(strike_vector) < eps)`, where eps is some small number

Reply:
> Looks generally fine to me; not sure whether this convention makes sense - but this is probably something that you guys have to decide!

The problem is that the strike direction is not constrained for a strike fault.
So I guess the best would be to use the direction of a neighbor fault element where the strike is constrained to define the strike orientation for the flat fault element.
But this is seems complicated to implement.
So the best is to define such an arbitrary convention to have an output, and then only visualise the absolute SR or absolute slip if part of the fault is flat.

> Does this actually work in practice? I'd expect something like (norm2(strike_vector) < eps), where eps is some small number

I think 0 can be accurately represented by a float or a double so there is not need to do norm2<eps.